### PR TITLE
refactor BreachCard component

### LINKED ISSUE #3423: Remove the BreachCard component in Vaccine Module Page
## The suggestion

Remove the Vaccine Module Page `BreachCard` component. It's just a style wrapper around BreachCard. We don't love it's existence.

**Why?**

Tidier.

## Describe alternatives you've considered

Do nothing. I suppose I could say it's low priority, but I am a frustratingly a stickler for tidy.

## Additional context

https://github.com/openmsupply/mobile/pull/3409#discussion_r566459793

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #3423 

The issue says to remove the component. I've refactored and removed from the `FridgeDetailPage` which may not be what was intended, but pleased me. The `FridgeDetailPage` is smaller and neater as a result

## Change summary

Created a new widget and update the `FridgeDetailPage`

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Check that the fridge page looks ok

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

File: src/widgets/BreachCard.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      headerText={headerText}
+      style={localStyles.card}
+      contentContainerStyle={{ alignItems: 'center', paddingBottom: 10 }}
+    >
+      <Text style={textStyle}>{text}</Text>
+      {icon}
+    </Paper>
+  );
+};
+
```
Comment (joshxg) [inline src/widgets/BreachCard.js:50]: I like this a lot, thanks.

I was wondering if we should actually take it further: the [mapStateToProps](https://github.com/openmsupply/mobile/blob/55ebde82acaf95465f393eb9859ed449db053c10/src/pages/FridgeDetailPage.js#L226-L276) is a bit of an issue as part of #3386 I was going to do this also- but was thinking to `connect` each breach card to redux directly, which would spread the time taken for the map state to props function and not front load it.

**Review by joshxg — APPROVED** [review]
Nice :)

A suggestion if you want to go further 🤣

**Comment by joshxg** [comment]
sorry I made conflicts cause I forgot to merge my PR earlier :(

File: src/widgets/BreachCard.js (hunk 1 above)
Comment (mark-prins) [reply src/widgets/BreachCard.js:50]: Nice! I've had a go, with this: c812b886

Wasn't sure how else to keep the shared elements in a single component while only referencing a specific part of the state. I think that this is quicker to load..?

**Review by joshxg — COMMENTED** [review]
I actually like this, but I think I was thinking something like:

```
const BreachCard = ({config}) => {
...
}


const stateToProps = (state, props) => {
	const {type} = props

	if (type === "HOT_CONSECUTIVE"){
		return { config: selectHotConsecutive(state), type: .. }
    }
}
```

**Comment by mark-prins** [comment]
ah, nice idea. thanks - will tidy up.

**Comment by mark-prins** [comment]
Thanks @joshxg : 6f2051b5 🤞

**Pull request merged**